File: leetcode/Python/robot-bounded-in-circle.py
class Solution:
    def isRobotBounded(self, instructions: str) -> bool:
        y, x, d = 0, 0, 0
        dy, dx = [1,0,-1,0], [0,-1,0,1]
        for i in range(4):
            for ins in instructions:
                if ins == 'L':
                    d = (d+1)%4
                elif ins == 'R':
                    d = (d+3)%4
                else:
                    y, x = y+dy[d], x+dx[d]
            if y == 0 and x == 0 and d == 0:
                return True
        return False

File: leetcode/C++/robot-bounded-in-circle.cpp
class Solution {
public:
    bool isRobotBounded(string instructions) {
        int y=0, x=0, d=0;
        int dy[]={1,0,-1,0}, dx[]={0,-1,0,1};
        for(int i=0; i<4; ++i){
            for(char &ins : instructions){
                if(ins == 'L') d=(d+1)%4;
                else if(ins == 'R') d=(d+3)%4;
                else {y+=dy[d]; x+=dx[d];}
            }
            if(y==0 && x==0 && d==0) return true;
        }
        return false;
    }
};